src/react/components.js

```diff
@@ -156,15 +128,17 @@ export function createClassInstance(
     }
   }
   // assign state
-  Properties.Set(realm, instance, "state", createAbstractObject(realm, "this.state", null), true);
+  Properties.Set(realm, instance, "state", createAbstractObject(realm, "this.state"), true);
   // assign refs
-  Properties.Set(realm, instance, "refs", createAbstractObject(realm, "this.refs", null), true);
+  Properties.Set(realm, instance, "refs", createAbstractObject(realm, "this.refs"), true);
   // assign props
   Properties.Set(realm, instance, "props", props, true);
   // assign context
   Properties.Set(realm, instance, "context", context, true);
   // enable serialization to support simple instance variables properties
   instance.refuseSerialization = false;
   // return the instance in an abstract object
-  return createAbstractObject(realm, "this", null, instance);
+  let value = createAbstractObject(realm, "this", instance);
+  invariant(value instanceof AbstractObjectValue);
```

Comment:
The type annotation of createAbstractObject should already guarantee this.

Reply:
Flow complains for me :(

Reply:
Then create an issue for the Flow team and use FlowFixMe to suppress the error and to reference the issue.